src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java

```diff
@@ -545,7 +557,33 @@ private void forEachBoundFolder(BiConsumer<WorkspaceFolderWrapper, WorkspaceFold
         boundFolderConsumer.accept(f, settings);
       }
     });
+  }
+
+  public void handleEvents(ServerEvent event){
+    if(event instanceof TaintVulnerabilityRaisedEvent){
+      var filePathFromEvent = ((TaintVulnerabilityRaisedEvent) event).getMainLocation().getFilePath();
+      updateTaintIssueCacheFromStorageForFile(URI.create(filePathFromEvent));
```

Comment:
I'm not sure we can convert the file path to a URI directly. You probably need to use `ProjectBindingManager.serverPathToFileUri`